src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.test.ts

```diff
@@ -146,6 +147,23 @@ describe('migrationsStateActionMachine', () => {
       }
     `);
   });
+
+  // see https://github.com/elastic/kibana/issues/98406
+  it('correctly logs state transition when using a logger adapter', async () => {
```

Comment:
Does it test that anything logged at all?

Reply:
The previous test is asserting that the log output is correct, but you're right, given the bug at stake here, I doesn't cost much to assert the underlying logger was properly called. Will adapt.